lib/Client.js

```diff
@@ -1042,39 +1054,107 @@ class Client extends EventEmitter {
      * @arg {String} options.name The thread channel name
+     * @arg {Number} [options.rateLimitPerUser] The time in seconds a user has to wait before sending another message (0-21600) (does not affect bots or users with manageMessages/manageChannel permissions)
+     * @arg {String} [options.reason] The reason to be displayed in audit logs
      * @returns {Promise<NewsThreadChannel | PublicThreadChannel>}
      */
     createThreadWithMessage(channelID, messageID, options) {
         return this.requestHandler.request("POST", Endpoints.THREAD_WITH_MESSAGE(channelID, messageID), true, {
+            auto_archive_duration: options.autoArchiveDuration,
             name: options.name,
-            auto_archive_duration: options.autoArchiveDuration
+            rate_limit_per_user: options.rateLimitPerUser,
+            reason: options.reason
         }).then((channel) => Channel.from(channel, this));
     }
 
     /**
-     * Create a thread without an existing message
+     * [DEPRECATED] Create a thread without an existing message. Use `createThread` instead
      * @arg {String} channelID The ID of the channel
      * @arg {Object} options The thread options
-     * @arg {Number} options.autoArchiveDuration Duration in minutes to automatically archive the thread after recent activity, either 60, 1440, 4320 or 10080
+     * @arg {Number} [options.autoArchiveDuration] Duration in minutes to automatically archive the thread after recent activity, either 60, 1440, 4320 or 10080
      * @arg {Boolean} [options.invitable] Whether non-moderators can add other non-moderators to the thread (private threads only)
      * @arg {String} options.name The thread channel name
-     * @arg {Number} [options.type] The channel type of the thread to create. It is recommended to explicitly set this property as this will be a required property in API v10
-     * @returns {Promise<PrivateThreadChannel>}
+     * @arg {Number} [options.rateLimitPerUser] The time in seconds a user has to wait before sending another message (0-21600) (does not affect bots or users with manageMessages/manageChannel permissions)
+     * @arg {String} [options.reason] The reason to be displayed in audit logs
+     * @arg {Number} [options.type] The channel type of the thread to create. This property defaults to private thread (12), but it is recommended to explicitly set this property as this will be a required property in API v10
+     * @returns {Promise<NewsThreadChannel | PublicThreadChannel | PrivateThreadChannel>}
      */
     createThreadWithoutMessage(channelID, options) {
-        return this.requestHandler.request("POST", Endpoints.THREAD_WITHOUT_MESSAGE(channelID), true, {
-            auto_archive_duration: options.autoArchiveDuration,
-            invitable: options.invitable,
-            name: options.name,
-            type: options.type
-        }).then((channel) => Channel.from(channel, this));
+        return this.createThread.call(this, channelID, options);
```

Comment:
```suggestion
        return this.requestHandler.request("POST", Endpoints.THREAD_WITHOUT_MESSAGE(channelID), true, {
            auto_archive_duration: options.autoArchiveDuration,
            invitable: options.invitable,
            name: options.name,
            type: options.type
        }).then((channel) => Channel.from(channel, this));
```
Preserve existing functionality